pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -115,3 +194,39 @@ func getInfraMachineFromProvider(platform configv1.PlatformType) (client.Object,
 		return nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
 	}
 }
+
+// shouldMirrorCAPIMachine takes a CAPI machine and determines if there should
+// be a MAPI mirror, it returns true only if:
+//
+// 1. The CAPI Machine is owned by a CAPI MachineSet,
+// 2. That CAPI MachineSet has a MAPI MachineSet Mirror.
+func (r *MachineSyncReconciler) shouldMirrorCAPIMachine(ctx context.Context, machine *capiv1beta1.Machine, logger logr.Logger) (bool, error) {
+	// Checks if owner ref is present
+	if len(machine.ObjectMeta.OwnerReferences) > 0 {
+		// If present, checks if owner ref points to CAPI MachineSet
+		for _, ref := range machine.ObjectMeta.OwnerReferences {
+			if ref.Kind == "MachineSet" && ref.APIVersion == "cluster.x-k8s.io/v1beta1" {
```

Comment:
How does this code block change if you were to invert this? Is it cleaner?
```suggestion
			if ref.Kind != "MachineSet" || ref.APIVersion != "cluster.x-k8s.io/v1beta1" {
```

Reply:
Ah yep nice, gets rid of the nasty nested ifs, ty! :)